lib/diagnostic.rb

```diff
@@ -7,38 +7,46 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.each.length.max ...?
   end
 
+# I'll need to review this but I understand we're trying to find the longest word in the list.
+# We would need to iterate through each item in the array and determine the length of each word.
+# Then find and return the maximum number based on the longest word of the list.
+
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.select(&:odd?).any?
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    numbers.select(&:even?).all?
```

Comment:
Here you can't use `select` first or it messes with the results. This works though:

`numbers.all?(&:even?)`